esphome/components/hlw8032/hlw8032.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/hal.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/uart/uart.h"
+
+#include <cinttypes>
+
+namespace esphome {
+namespace hlw8032 {
+
+class HLW8032Component : public Component, public uart::UARTDevice {
+ public:
+  void loop() override;
+  void dump_config() override;
+
+  void set_current_resistor(float current_resistor) { current_resistor_ = current_resistor; }
```

Comment:
> All uses of class members and member functions should be prefixed with this-> to distinguish them from global functions/variables.

https://developers.esphome.io/contributing/code/#c-code-style

```suggestion
  void set_current_resistor(float current_resistor) { this->current_resistor_ = current_resistor; }
```